src/arithmetic.jl

```diff
@@ -407,89 +365,70 @@ function /{T<:Integer, S<:NumberNotSeries}(a::Taylor1{Rational{T}}, b::S)
 end
 
 for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
-    @eval begin
-        /{T<:NumberNotSeries}(a::$T{T}, b::T) = a * inv(b)
-        function /{T<:NumberNotSeries,S<:NumberNotSeries}(a::$T{T}, b::S)
-            R = promote_type(T,S)
-            return convert($T{R}, a) * inv(convert(R, b))
-        end
-        /{T<:NumberNotSeries}(a::$T, b::T) = a * inv(b)
+
+    @eval /{T<:NumberNotSeries}(a::$T{T}, b::T) = a * inv(b)
+
+    @eval function /{T<:NumberNotSeries,S<:NumberNotSeries}(a::$T{T}, b::S)
```

Comment:
Any particular reason for splitting up the `@eval begin...end` into separate `@eval` here?